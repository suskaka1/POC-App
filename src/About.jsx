import React, { Component } from "react";
 
class About extends Component {
  render() {
    return (
      <div>
        <h2>About ReactJS </h2>
        <div className="ui divider"></div>
        <div>
          <h3>ReactJS</h3><p>React is a front-end library developed by Facebook</p>
          <div className="ui divider"></div>
          <h3>Redux</h3><p>Redux is quite an excellent State Managment Framework usually used with React.js library. In Single Page Application, data management at client side is far more complicated than just imagine. However, In React.js state management is possible, but when the application gets bigger and bigger, unwanted errors and data changes are detected, and which module has changed which state and which view is updated, all these matters get complex, and we feel like, we trapped in our application. Facebook gives the solution. Its developer has created one State management pattern called Flux.</p>
          <div className="ui divider"></div>
          <h3>Cucumber</h3><p>Cucumber is a Behaviour Driven Development (BDD) tool that allows you to write software requirements as specifications in a human-readable format, and use those specifications to run tests that make sure that the software does what is expected.</p>
          <div className="ui divider"></div>
<h3>Jest</h3><p>Jest is a delightful JavaScript Testing Framework with a focus on simplicity.</p>
        </div>
      </div>
    );
  }
}
 
export default About;