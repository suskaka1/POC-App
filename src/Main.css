.menu-left-part
{
    background-color: white;
    float: left;
    padding: 45px 45px 0px 50px;
    width: 29.2%;
    height: 560px;
    font-weight: bold;
    font-size: 28px;
}

h1 a {
    color: #CC0000;
    text-decoration: none;
}

h1 a:focus {
    color: #666666;
    text-decoration: none;
}

.detail-right-part
{
    background-color: white;
    margin: 0px 0px 0px 20px;
    float: left;
    width: 65%;
    padding: 25px 10px 10px 20px;
    height: 560px;
}

.menu-postion{
    margin: 0% 0% 0% 10%;
    text-align: left;
    width: 80%;
}