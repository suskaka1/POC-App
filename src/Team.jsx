import React, { Component } from "react";
class Team extends Component {

  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isLoaded: false,
      items: []
    };
  }

  componentDidMount() {
    var url = "https://jsonplaceholder.typicode.com/users";
    fetch(url)
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            items: result
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      )
  }

  render() {
    const { error, isLoaded, items } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>;
    } else if (!isLoaded) {
      return <div>Loading...</div>;
    } else {
      return (
        <div>
          <h2>Meet the Team </h2>
          <div className="ui divider"></div>
          {/* <h4>Amit Kumar</h4> */}
          {/* <ul> */}
            {
              
              items.map(item => (
                
                <h3 key={item.id}>
                  {/* {item.name} - {item.email} */}
                  {/* <img src={myImage} className="imagesize ui circular image" /> */}
                  {item.id}. {item.name}
                </h3>
              ))}
          {/* </ul> */}
        </div>

      );
    }
  }
}

export default Team;