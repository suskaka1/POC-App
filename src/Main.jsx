import React, { Component } from "react";
import {
  Route,
  NavLink,
  HashRouter
} from "react-router-dom";
import Dashboard from "./Dashboard";
import About from "./About";
import Team from "./Team";
import './Main.css';
import Weather from "./Weather";
import NewTeamMember from "./NewTeamMember";


class Main extends Component {
  render() {
    return (
      <HashRouter>
        <div className='menu-postion'>
          <div className='menu-left-part'>
            <h1><NavLink to="/dashboard">Dashboard</NavLink></h1>
            <div className="ui divider"></div>
            <h1  id="about"><NavLink to="/about">About</NavLink></h1>
            <div className="ui divider"></div>
            <h1><NavLink to="/team">Team</NavLink></h1>
            <div className="ui divider"></div>
            <h1><NavLink to="/addmember">New Team Member</NavLink></h1>
            <div className="ui divider"></div>
            <h1><NavLink to="/weather">Weather Report</NavLink></h1>
          </div>

          <div className='detail-right-part'>
            <Route path="/dashboard" component={Dashboard} />
            <Route path="/about" component={About} />
            <Route path="/team" component={Team} />
            <Route path="/weather" component={Weather} />
            <Route path="/addmember" component={NewTeamMember} />
          </div>
        </div>
      </HashRouter>
    );
  }
}

export default Main;