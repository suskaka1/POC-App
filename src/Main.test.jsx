import React from 'react';
import Main from './Main';
import About from './About';
import Team from "./Team";
import { cleanup } from 'react-testing-library';
import { shallow, mount } from 'enzyme';
import {
  Route,
  NavLink,
  HashRouter
} from "react-router-dom";

describe("Main", () => {
  it("should render my right side menu link page", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });


  
  // it('should update the page when select a menu list', () => {
  //   const wrapper = shallow(<Main />);
  //   const aboutPagelink = wrapper.find(NavLink).at(1);
  //   aboutPagelink.simulate('click');
  //   expect(window.location.hash).toBe('#/about');
  // })
});