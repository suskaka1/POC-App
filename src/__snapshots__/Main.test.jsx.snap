// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render my right side menu link page 1`] = `ShallowWrapper {}`;
